rocAL/rocAL/source/readers/image/external_source_reader.cpp

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <cassert>
+#include <algorithm>
+// #include <commons.h>
+#include "external_source_reader.h"
+#include <boost/filesystem.hpp>
+
+namespace filesys = boost::filesystem;
+
+ExternalSourceReader::ExternalSourceReader()
+{
+    _curr_file_idx = 0;
+    _current_file_size = 0;
+    _current_fPtr = nullptr;
+    _loop = false;
+    _file_id = 0;
+    _shuffle = false;
+    _file_count_all_shards = 0;
+    _loop = false;    // loop not supported for external source
+    _end_of_sequence = false;
+}
+
+// return batch_size() for count_items unless end_of_sequence has been signalled.
+unsigned ExternalSourceReader::count_items()
+{
+    if (_mode == FileMode::FILENAME) {
+        if (_end_of_sequence && _file_names_q.empty()) {
+        return 0;
+        }
+    } else {
+        if (_end_of_sequence && _images_data_q.empty()) {
+        return 0;
+        }
+    }
+    return _batch_count;
+}
+
+Reader::Status ExternalSourceReader::initialize(ReaderConfig desc)
+{
+    auto ret = Reader::Status::OK;
+    _folder_path = desc.path();
+    _file_id = 0;
+    _shard_id = desc.get_shard_id();
+    _shard_count = desc.get_shard_count();
+    _batch_count = desc.get_batch_size();
+    _shuffle = desc.shuffle();
+    _loop = desc.loop();
+    _mode = desc.mode();
+    _end_of_sequence = false;
+    _file_data.reserve(_batch_count);
+    return ret;
+}
+
+void ExternalSourceReader::incremenet_read_ptr()
+{
+    _read_counter++;
+    // _curr_file_idx++;
+        _curr_file_idx = (_curr_file_idx + 1) % _batch_count;
+}
+
+size_t ExternalSourceReader::open()
+{
+    if (_mode == FileMode::FILENAME) {
+        std::string next_file_name;
+        bool ret = pop_file_name(next_file_name);   // Get next file name: blocking call, will wait till next file is received from external source
+        if (_end_of_sequence && !ret) // shobi check this
```

Comment:
Here there is a name tag associated. Please check if this case is addressed

Reply:
Yes. It is addressed